### ISSUE TITLE
False negative when pattern matching with a non-class

### ISSUE THREAD

**Issue by lihu-zhong** [astral-sh/ty#2592]
### Summary

This may be a duplicate of #887 but I didn't see this specific case mentioned, so I'm throwing it out here. Feel free to close if another issue isn't necessary.

**Observed behavior**

The following code passes `ty` but throws a `TypeError` at runtime:

```python
foo = "foo"

match "bar":
  case foo():
    ...
```

**Expected behavior**

`ty` should raise an error when a non-type is used in a class pattern when matching. Both `mypy` and `pyright` catch this.

### Version

0.0.10

**Comment by AlexWaygood** [comment]
Thanks -- agreed; we're definitely missing a check here

**Comment by abhijeetbodas2001** [comment]
Hi, can I work on this? We'll need a new diagnostic here right?

**Comment by AlexWaygood** [comment]
Yes, go for it! And, yes, I think a new diagnostic makes sense here
